File: csharp/resistor-color/ResistorColor.cs
﻿using System;
using System.Collections.Generic;
public static class ResistorColor
{
    static string[] colors = new string[]
    {
        "black", "brown", "red", "orange",
        "yellow", "green", "blue", "violet",
        "grey", "white"
    };
    public static int ColorCode(string color) => Array.IndexOf(colors, color);

    public static string[] Colors() => colors;
}

File: elixir/resistor-color/lib/resistor_color.ex
defmodule ResistorColor do
  @moduledoc false
   @colors ~w(black brown red orange yellow green blue violet grey white)

  @spec colors() :: list(String.t())
  def colors do
    @colors
  end

  @spec code(String.t()) :: integer()
  def code(color), do: @colors |> Enum.find_index(&(&1 == color))
end
